src/libs/sfdk/device.cpp

```diff
@@ -115,6 +119,40 @@ Device::MachineType Device::machineType() const
     return d_func()->machineType;
 }
 
+void Device::importPublicGpgKey(const QString &id,
+        const QObject *context, const Functor<bool, QString> &functor)
+{
+    QString errorString;
+    QTC_ASSERT(isGpgAvailable(&errorString),
+            QTimer::singleShot(0, context, std::bind(functor, false, errorString));
+            return );
+
+    const QPointer<const QObject> context_{context};
+    SigningUtils::exportPublicKey(id, Sdk::instance(),
+            [=](bool ok, const QByteArray &keyData, const QString &exportError) {
+        if (!ok) {
+            callIf(context_, functor, ok, exportError);
+            return;
+        }
+
+        QStringList arguments;
+        arguments.append("-");
+
+        auto runner = std::make_unique<RemoteProcessRunner>(sshParameters(),
+                "sdk-deploy-key",
```

Comment:
I think it would be better to add an argument, such as `-k --key and -f --file` instead of `-`, to use via ssh
@martyone ?

Reply:
The acceptance of either file name or '-' as a positional argument follows the conventions used by coreutils, so I would keep it as it. We could make the argument optional though, reading from stdin whether it is '-' or no file is passed at all. And allow to pass multiple keys at once . Not necessarily, just an idea.